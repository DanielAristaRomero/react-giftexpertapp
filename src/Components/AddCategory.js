import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('');

    const handelInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategorias(cats => [inputValue, ...cats]);
            setInputValue('');
        }

        
    };

    return (
        <form onSubmit = {handleSubmit}>
            <input 
                className = 'buscador'
                type = 'text'
                value = {inputValue}
                onChange = {handelInputChange}
                />
        </form>
    );
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired,
}