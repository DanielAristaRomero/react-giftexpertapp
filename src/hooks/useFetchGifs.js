import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (categoria) => {
    
    const [state, setState] = useState({
        data: [], 
        loading: true,
    }); 

    useEffect( () => {

        getGifs(categoria)
            .then(imgs => {

                setState({
                    data: imgs,
                    loading: false,
                });
            })

    }, [categoria]);

    // useEffect( () => {
    //     getGifs(categoria)
    //         .then(imgs => setImages(imgs));
    // }, [categoria]);


    return state;
}


