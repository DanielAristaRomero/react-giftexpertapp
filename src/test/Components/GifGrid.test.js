import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    
    const category = 'Pruebas';
    
    test('debe de mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'abc',
            url: 'https://localhost',
            title: 'cualquier cosa',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    
});
