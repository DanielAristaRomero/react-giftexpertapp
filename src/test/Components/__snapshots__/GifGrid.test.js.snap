// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="categoria"
  />
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://localhost"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="categoria"
  />
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
